cli/script/command-executor.ts

```diff
@@ -1644,3 +1648,77 @@ function getSdk(accessKey: string, headers: Headers, customServerUrl: string): A
 
   return sdk;
 }
+
+function createPatch(command: cli.ICreatePatchCommand): Promise<void> {
+  return Q.Promise<void>((resolve, reject) => {
+    const scriptPath = path.join(__dirname, "patch-scripts", "create-patch.sh");
+    
+    // Normalize the patch directory path
+    let patchDir = command.patchFile;
+    if (!path.isAbsolute(patchDir)) {
+      patchDir = path.resolve(process.cwd(), patchDir);
+    }
+    
+    // Create the directory if it doesn't exist
+    try {
+      fs.mkdirSync(patchDir, { recursive: true });
+    } catch (err) {
+      if (err.code !== 'EEXIST') {
+        reject(new Error(`Failed to create directory: ${err.message}`));
+        return;
+      }
+    }
+    
+    const args = [command.oldBundle, command.newBundle, patchDir, false];
+    
+    log("==Input Arguments==");
+    log(`Old bundle: ${command.oldBundle}`);
+    log(`New bundle: ${command.newBundle}`);
+    log(`Patch directory: ${patchDir}`);
+    
+    const child = childProcess.spawn("bash", [scriptPath, ...args], {
+      stdio: "inherit",
+      cwd: process.cwd()
+    });
+    
+    child.on("close", (code: number) => {
+      if (code === 0) {
+        log("Patch file created successfully!");
+        resolve();
+      } else {
+        reject(new Error(`Patch creation failed with exit code ${code}`));
+      }
+    });
+    
+    child.on("error", (error: Error) => {
+      reject(new Error(`Failed to start patch creation: ${error.message}`));
```

Comment:
```suggestion
      reject(new Error(`Failed to start patch creation: ${error.message}`));
```

Are these Exceptions automatically logged?

Reply:
Yupp, there is a central catch block.